web3swift/src/Account/EthereumAccount.swift

```diff
@@ -9,173 +9,107 @@ import Foundation
     }
 
-    required public init(keyStorage: EthereumSingleKeyStorageProtocol, logger: Logger? = nil) throws {
-        self.logger = logger ?? Logger(label: "web3.swift.eth-account")
-        do {
-            let data = try keyStorage.loadPrivateKey()
-            self.privateKeyData = data
-            self.publicKeyData = try KeyUtil.generatePublicKey(from: data)
-        } catch {
-            throw EthereumAccountError.loadAccountError
-        }
-    }
-
-    required public init(addressString: String, keyStorage: EthereumMultipleKeyStorageProtocol, keystorePassword password: String, logger: Logger? = nil) throws {
-        self.logger = logger ?? Logger(label: "web3.swift.eth-account")
-        do {
-            let address = EthereumAddress(addressString)
-            let decodedKey = try keyStorage.loadAndDecryptPrivateKey(for: address, keystorePassword: password)
-            self.privateKeyData = decodedKey
-            self.publicKeyData = try KeyUtil.generatePublicKey(from: decodedKey)
-        } catch {
-            self.logger.warning("Error loading key data: \(error)")
-            throw EthereumAccountError.loadAccountError
-        }
-    }
-
-    required public init(addressString: String, keyStorage: EthereumMultipleKeyStorageProtocol, logger: Logger? = nil) throws {
-        self.logger = logger ?? Logger(label: "web3.swift.eth-account")
-        do {
-            let address = EthereumAddress(addressString)
-            let data = try keyStorage.loadPrivateKey(for: address)
-            self.privateKeyData = data
-            self.publicKeyData = try KeyUtil.generatePublicKey(from: data)
-        } catch {
-            throw EthereumAccountError.loadAccountError
-        }
-    }
-
-    public static func create(addingTo keyStorage: EthereumMultipleKeyStorageProtocol, keystorePassword password: String) throws -> EthereumAccount {
+    public static func create(settingTo keyStorage: EthereumKeyStorageProtocol) async throws -> EthereumAccount {
         guard let privateKey = KeyUtil.generatePrivateKeyData() else {
             throw EthereumAccountError.createAccountError
         }
 
         do {
-            try keyStorage.encryptAndStorePrivateKey(key: privateKey, keystorePassword: password)
             let publicKey = try KeyUtil.generatePublicKey(from: privateKey)
-            let address = KeyUtil.generateAddress(from: publicKey).value
-            return try self.init(addressString: address, keyStorage: keyStorage, keystorePassword: password)
+            let address = KeyUtil.generateAddress(from: publicKey)
+            try await keyStorage.storePrivateKey(key: privateKey, with: address)
+            return self.init(address: address, keyStorage: keyStorage)
         } catch {
             throw EthereumAccountError.createAccountError
         }
     }
 
-    public static func create(replacing keyStorage: EthereumSingleKeyStorageProtocol, keystorePassword password: String) throws -> EthereumAccount {
```

Comment:
These initializers would still be needed, to allow for single or multiple key storage initialization.